src/streamlink/plugin/plugin.py

```diff
@@ -483,5 +518,22 @@ def clear_cookies(self, cookie_filter=None):
 
         return removed
 
+    def input_ask(self, prompt):
+        if self._user_input_requester:
```

Comment:
if someone uses **streamlink as a subprocess**,
there might be no console to actually type the value in.

There might be a cancel button if the set up is nicely _(like in https://github.com/streamlink/streamlink-twitch-gui)_,
if not it will run forever in the background until someone cancels it.

---

**_some solutions_**

1. Add an option so this can be disabled,
so it will always return an Error or it will be skipped and the Plugin must handle the Error.

2. Timeout the prompt after X minutes.

Reply:
I guess that is an issue that exists already in `Console.ask` and `Console.askpass`. I think the best thing to do in this case is test to see if there is a `tty` available, and if not raise an exception.

Reply:
not sure if this is how it should work with `--retry-open`

or if this is an issue

```
Found matching plugin twitch for URL twitch.tv/...
Plugin specific arguments:
--twitch-disable-hosting=True (disable_hosting)
User input error: no TTY available
Waiting for streams, retrying every 1.0 second(s)
User input error: no TTY available
User input error: no TTY available
User input error: no TTY available
User input error: no TTY available
...
```

it won't stop

Reply:
Probably shouldn’t print that so much, I guess it’s a side effect - which input is it waiting for though?

Reply:
this is my testcase with https://github.com/streamlink/streamlink-twitch-gui git version

```diff
diff --git a/src/streamlink/plugins/twitch.py b/src/streamlink/plugins/twitch.py
index d42cba2..65bced6 100644
--- a/src/streamlink/plugins/twitch.py
+++ b/src/streamlink/plugins/twitch.py
@@ -631,6 +631,7 @@ class Twitch(Plugin):
         return streams
 
     def _get_streams(self):
+        self.input_ask('123')
         if self.video_id:
             if self.video_type == "v":
                 return self._get_hls_streams("video")
```

Reply:
Ah ok. Cool, I will take a look - I guess a effect of retry open is that you will get that error repeatedly. Not sure if it can be avoided, if you need to give that input and you cannot pass it via the command line then maybe the plugin cannot run - if it can run without the input then the plugin should handle the error... maybe.

Reply:
I added a `FatalPluginError`, which by default cannot be recovered from - unless the Plugin overrides this behaviour (see https://github.com/streamlink/streamlink/pull/1827#issuecomment-398707513 for an example).